frontend/src/components/FindMatch.js

```diff
@@ -53,6 +55,10 @@ const FindMatch = () => {
     }
   }
 
+  window.onbeforeunload = (event) => {
+    deleteMatch(user._id, matchingSocket.id, difficulty) // reset matching service
+  }
```

Comment:
Just a small note to be aware of: The body of a React functional component gets executed every render, which means that this event handler gets attached every time the component re-renders. Generally, event handlers are set up in this pattern instead:
```
// event listener is only created and registered once in the component's entire lifecycle (i.e. when it is mounted)
useEffect(() => {
  window.addEventListener('beforeunload', <handler here>)

  // event listener is cleaned up when the component unmounts to prevent bugs and leakage
  return () => {
    window.removeEventListener('beforeunload', <handler here>)
  }
}, [])
```